azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/certificate/PfxCertReader.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.keyvault.spring.certificate;
+
+import org.springframework.core.io.Resource;
+
+import java.io.*;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Enumeration;
+
+public class PfxCertReader implements KeyCertReader {
+    private String CERT_NOT_FOUND = "Cert file %s not found.";
+    private String CERT_READ_FAILURE = "Failed to read cert file %s.";
+
+    @Override
+    public KeyCert read(Resource resource, String password) {
+        try (InputStream inputStream = resource.getInputStream()) {
+            KeyCert keyCert = null;
+
+            final KeyStore store = KeyStore.getInstance("pkcs12", "SunJSSE");
```

Comment:
Not sure whether this is heavy, but it's better to put it into field to reuse.

Reply:
Might be reused for many times, moved store to instance variable.